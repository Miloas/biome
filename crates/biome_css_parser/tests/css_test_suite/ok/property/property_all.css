div {
    /* Global values */
    all: inherit;
    all: initial;
    all: unset;
    all: revert;
    all: revert-layer;
}