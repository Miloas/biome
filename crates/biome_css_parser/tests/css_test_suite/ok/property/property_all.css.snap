---
source: crates/biome_css_parser/tests/spec_test.rs
expression: snapshot
---

## Input

```css
div {
    /* Global values */
    all: inherit;
    all: initial;
    all: unset;
    all: revert;
    all: revert-layer;
}
```


## AST

```
CssRoot {
    bom_token: missing (optional),
    rules: CssRuleList [
        CssRule {
            prelude: CssSelectorList [
                CssCompoundSelector {
                    nesting_selector_token: missing (optional),
                    simple_selector: CssTypeSelector {
                        namespace: missing (optional),
                        ident: CssIdentifier {
                            value_token: IDENT@0..4 "div" [] [Whitespace(" ")],
                        },
                    },
                    sub_selectors: CssSubSelectorList [],
                },
            ],
            block: CssDeclarationListBlock {
                l_curly_token: L_CURLY@4..5 "{" [] [],
                declarations: CssDeclarationList [
                    CssDeclaration {
                        property: CssAllProperty {
                            name: CssIdentifier {
                                value_token: IDENT@5..37 "all" [Newline("\n"), Whitespace("    "), Comments("/* Global values */"), Newline("\n"), Whitespace("    ")] [],
                            },
                            colon_token: COLON@37..39 ":" [] [Whitespace(" ")],
                            value: CssWideKeyword {
                                value: INHERIT_KW@39..46 "inherit" [] [],
                            },
                        },
                        important: missing (optional),
                    },
                    SEMICOLON@46..47 ";" [] [],
                    CssDeclaration {
                        property: CssAllProperty {
                            name: CssIdentifier {
                                value_token: IDENT@47..55 "all" [Newline("\n"), Whitespace("    ")] [],
                            },
                            colon_token: COLON@55..57 ":" [] [Whitespace(" ")],
                            value: CssWideKeyword {
                                value: INITIAL_KW@57..64 "initial" [] [],
                            },
                        },
                        important: missing (optional),
                    },
                    SEMICOLON@64..65 ";" [] [],
                    CssDeclaration {
                        property: CssAllProperty {
                            name: CssIdentifier {
                                value_token: IDENT@65..73 "all" [Newline("\n"), Whitespace("    ")] [],
                            },
                            colon_token: COLON@73..75 ":" [] [Whitespace(" ")],
                            value: CssWideKeyword {
                                value: UNSET_KW@75..80 "unset" [] [],
                            },
                        },
                        important: missing (optional),
                    },
                    SEMICOLON@80..81 ";" [] [],
                    CssDeclaration {
                        property: CssAllProperty {
                            name: CssIdentifier {
                                value_token: IDENT@81..89 "all" [Newline("\n"), Whitespace("    ")] [],
                            },
                            colon_token: COLON@89..91 ":" [] [Whitespace(" ")],
                            value: CssWideKeyword {
                                value: REVERT_KW@91..97 "revert" [] [],
                            },
                        },
                        important: missing (optional),
                    },
                    SEMICOLON@97..98 ";" [] [],
                    CssDeclaration {
                        property: CssAllProperty {
                            name: CssIdentifier {
                                value_token: IDENT@98..106 "all" [Newline("\n"), Whitespace("    ")] [],
                            },
                            colon_token: COLON@106..108 ":" [] [Whitespace(" ")],
                            value: CssWideKeyword {
                                value: REVERT_LAYER_KW@108..120 "revert-layer" [] [],
                            },
                        },
                        important: missing (optional),
                    },
                    SEMICOLON@120..121 ";" [] [],
                ],
                r_curly_token: R_CURLY@121..123 "}" [Newline("\n")] [],
            },
        },
    ],
    eof_token: EOF@123..123 "" [] [],
}
```

## CST

```
0: CSS_ROOT@0..123
  0: (empty)
  1: CSS_RULE_LIST@0..123
    0: CSS_RULE@0..123
      0: CSS_SELECTOR_LIST@0..4
        0: CSS_COMPOUND_SELECTOR@0..4
          0: (empty)
          1: CSS_TYPE_SELECTOR@0..4
            0: (empty)
            1: CSS_IDENTIFIER@0..4
              0: IDENT@0..4 "div" [] [Whitespace(" ")]
          2: CSS_SUB_SELECTOR_LIST@4..4
      1: CSS_DECLARATION_LIST_BLOCK@4..123
        0: L_CURLY@4..5 "{" [] []
        1: CSS_DECLARATION_LIST@5..121
          0: CSS_DECLARATION@5..46
            0: CSS_ALL_PROPERTY@5..46
              0: CSS_IDENTIFIER@5..37
                0: IDENT@5..37 "all" [Newline("\n"), Whitespace("    "), Comments("/* Global values */"), Newline("\n"), Whitespace("    ")] []
              1: COLON@37..39 ":" [] [Whitespace(" ")]
              2: CSS_WIDE_KEYWORD@39..46
                0: INHERIT_KW@39..46 "inherit" [] []
            1: (empty)
          1: SEMICOLON@46..47 ";" [] []
          2: CSS_DECLARATION@47..64
            0: CSS_ALL_PROPERTY@47..64
              0: CSS_IDENTIFIER@47..55
                0: IDENT@47..55 "all" [Newline("\n"), Whitespace("    ")] []
              1: COLON@55..57 ":" [] [Whitespace(" ")]
              2: CSS_WIDE_KEYWORD@57..64
                0: INITIAL_KW@57..64 "initial" [] []
            1: (empty)
          3: SEMICOLON@64..65 ";" [] []
          4: CSS_DECLARATION@65..80
            0: CSS_ALL_PROPERTY@65..80
              0: CSS_IDENTIFIER@65..73
                0: IDENT@65..73 "all" [Newline("\n"), Whitespace("    ")] []
              1: COLON@73..75 ":" [] [Whitespace(" ")]
              2: CSS_WIDE_KEYWORD@75..80
                0: UNSET_KW@75..80 "unset" [] []
            1: (empty)
          5: SEMICOLON@80..81 ";" [] []
          6: CSS_DECLARATION@81..97
            0: CSS_ALL_PROPERTY@81..97
              0: CSS_IDENTIFIER@81..89
                0: IDENT@81..89 "all" [Newline("\n"), Whitespace("    ")] []
              1: COLON@89..91 ":" [] [Whitespace(" ")]
              2: CSS_WIDE_KEYWORD@91..97
                0: REVERT_KW@91..97 "revert" [] []
            1: (empty)
          7: SEMICOLON@97..98 ";" [] []
          8: CSS_DECLARATION@98..120
            0: CSS_ALL_PROPERTY@98..120
              0: CSS_IDENTIFIER@98..106
                0: IDENT@98..106 "all" [Newline("\n"), Whitespace("    ")] []
              1: COLON@106..108 ":" [] [Whitespace(" ")]
              2: CSS_WIDE_KEYWORD@108..120
                0: REVERT_LAYER_KW@108..120 "revert-layer" [] []
            1: (empty)
          9: SEMICOLON@120..121 ";" [] []
        2: R_CURLY@121..123 "}" [Newline("\n")] []
  2: EOF@123..123 "" [] []

```


